x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/email/EmailTemplate.java

```diff
@@ -110,16 +113,38 @@ public TextTemplate htmlBody() {
         if (subject != null) {
             builder.subject(engine.render(subject, model));
         }
-        if (textBody != null) {
-            builder.textBody(engine.render(textBody, model));
-        }
+
+        Set<String> warnings = new HashSet<>(1);
         if (attachments != null) {
             for (Attachment attachment : attachments.values()) {
                 builder.attach(attachment);
+                warnings.addAll(attachment.getWarnings());
             }
         }
+
```

Comment:
I just realized, that despite the current documenation, body is not actually required, will need to adjust this to always emit a warning even if the body is not defined.

Reply:
done in https://github.com/elastic/elasticsearch/pull/44460/commits/0acff82f309d0d4b56dc5f2f0da2eb849bef533a